### ISSUE TITLE
When PY3-only, use stdlib's tzinfo objects where viable

### ISSUE THREAD

**Issue by jbrockmendel** [dateutil/dateutil#867]
Specifically timezone.utc and fixed offsets.

**Comment by pganssle** [comment]
I'm guessing this is related to the parser?

I'm maybe -0 on this. This would be a backwards-incompatible change. The main benefit would be that in some situations the `stdlib`'s time zone objects will probably be faster (I have not benchmarked this so it's not necessarily true). On the other hand, there is no real plan to deprecate `tzoffset` or `tzutc`, and those classes have additional "fast path" methods like `is_ambiguous` designed to work with `tz.datetime_ambiguous` and `tz.datetime_exists`.

**Comment by jbrockmendel** [comment]
> I'm guessing this is related to the parser?

More about the equality semantics

**Comment by pganssle** [comment]
Yeah the equality semantics for time zones is kind of a mess, I dunno if we're going to solve that any time soon, though. We may be able to fiddle around and make it so `tzoffset` is equal to an equivalent `timezone` object, though. That said, it may be hard to make the hashing work out correctly if we do that.
